File: core/src/main/java/kandroid/data/FleetData.java
package kandroid.data;


import java.util.Date;
import java.util.List;

import kandroid.observer.POJO;
import kandroid.observer.RawData;
import kandroid.utils.Identifiable;

public class FleetData extends Data<FleetData.ApiDeck> implements Identifiable, RequestDataListner, ResponseDataListner {

    private boolean inSortie;

    public int getFleetID() {
        return data.api_id;
    }

    public String getName() {
        return data.api_name;
    }

    public int getExpeditionState() {
        return data.api_mission.get(0);
    }

    public int getExpeditionDestination() {
        return data.api_mission.get(1);
    }

    public Date getExpeditionTime() {
        return new Date(data.api_mission.get(2));
    }

    public List<Integer> getMembers() {
        return data.api_ship;
    }

    public List<ShipData> getMembersInstance() {
        throw new UnsupportedOperationException();
    }

    public List<ShipData> getMembersWithoutEscaped() {
        throw new UnsupportedOperationException();
    }

    public List<Integer> getEscapedShipList() {
        throw new UnsupportedOperationException();
    }

    public boolean isInSortie() {
        return inSortie;
    }

    public Date getConditionTime() {
        throw new UnsupportedOperationException();
    }

    public boolean isConditionTimeLocked() {
        throw new UnsupportedOperationException();
    }

    @Override
    public int getId() {
        return data.api_id;
    }

    @Override
    public void loadFromRequest(String apiName, RawData rawData) {
        switch (apiName) {
            case "api_req_nyukyo/start":
            case "api_req_nyukyo/speedchange":
                ShortenConditionTimer();
                break;
        }
    }

    private void ShortenConditionTimer() {
        //TODO
    }

    @Override
    public void loadFromResponse(String apiName, RawData rawData) {
        switch (apiName) {
            case "api_get_member/ndock":
                ShortenConditionTimer();
                break;
        }
    }

    //TODO

    public static class ApiDeck extends POJO {
        public int api_member_id;
        public int api_id;
        public String api_name;
        public String api_name_id;
        public String api_flagship;
        public List<Integer> api_mission;
        public List<Integer> api_ship;
    }
}

File: core/src/main/java/kandroid/data/FleetData.kt
package kandroid.data

import com.google.gson.JsonElement
import kandroid.newdata.JsonWrapper
import kandroid.observer.POJO
import kandroid.observer.RawData
import kandroid.utils.Identifiable
import java.util.*

class FleetData : JsonWrapper(), Identifiable {

    val isInSortie: Boolean = false

    val fleetID: Int
        get() = data.api_id

    val name: String
        get() = data.api_name

    val expeditionState: Int
        get() = data.api_mission!![0]

    val expeditionDestination: Int
        get() = data.api_mission!![1]

    val expeditionTime: Date
        get() = Date(data.api_mission!![2].toLong())

    val members: List<Int>
        get() = data.api_ship

    val membersInstance: List<ShipData>
        get() = throw UnsupportedOperationException()

    val membersWithoutEscaped: List<ShipData>
        get() = throw UnsupportedOperationException()

    val escapedShipList: List<Int>
        get() = throw UnsupportedOperationException()

    val conditionTime: Date
        get() = throw UnsupportedOperationException()

    val isConditionTimeLocked: Boolean
        get() = throw UnsupportedOperationException()

    override val id: Int
        get() = data.api_id

    fun loadFromRequest(apiName: String, rawData: RawData) {
        when (apiName) {
            "api_req_nyukyo/start", "api_req_nyukyo/speedchange" -> ShortenConditionTimer()
        }
    }

    private fun ShortenConditionTimer() {
        //TODO
    }

    override fun loadFromResponse(apiName: String, responseData: JsonElement) {
        when (apiName) {
            "api_get_member/ndock" -> ShortenConditionTimer()
        }
    }

    //TODO

    open class ApiDeck : POJO() {
        var api_member_id: Int = 0
        var api_id: Int = 0
        var api_name: String? = null
        var api_name_id: String? = null
        var api_flagship: String? = null
        var api_mission: List<Int>? = null
        var api_ship: List<Int>? = null
    }
}